lib/network/utils/RoundRobin.js

```diff
@@ -0,0 +1,128 @@
+const DEFAULT_STICKY_COUNT = 100;
+
+/**
+ * Shuffle an array in-place
+ *
+ * @param {Array} array - The array to shuffle
+ * @return {undefined}
+ */
+function shuffle(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+        const randIndex = Math.floor(Math.random() * (i + 1));
+        /* eslint-disable no-param-reassign */
+        const randIndexVal = array[randIndex];
+        array[randIndex] = array[i];
+        array[i] = randIndexVal;
+        /* eslint-enable no-param-reassign */
+    }
+}
+
+class RoundRobin {
+    /**
+     * @constructor
+     * @param {object[]} hostsList - list of hosts to query in
+     *   round-robin fashion
+     * @param {string} hostsList[].host - host name or IP address
+     * @param {number} hostsList[].port - port number to contact
```

Comment:
We can also be more flexible on the input format, and convert `host:port` strings as a common `{host, port}` JS object to the user (or the other way, but I tend to prefer the object form since it's more easily extensible).